### PR TITLE
docs: Add note on initializing structs with reference fields

### PR DESCRIPTION
**Pull request by lilenvy**
Fix: #7492

### CODE REVIEW

**Comment by medvednikov** [comment]
Assigning 0 to references is unsafe and temporary, it won't be allowed soon. V doesn't have nil.

**Comment by lilenvy** [comment]
Sorry I was just thinking a bit more about this problem, is this by any chance attempted to be solved via a sumtype?
For example:
```v
type Ref = &Node|none
struct Node {  next Ref }
```
IIRC I tried doing this but I get `sum type cannot hold a reference type` so I'm not sure what the expected value should be.

**Comment by spytheman** [comment]
@medvednikov in my opinion, that documentation PR should be merged. 
It is very clear and explains a common question well (how to make a linked list in V).
Until we have a better solution, I think it will serve well new V users.

**Comment by lilenvy** [comment]
thanks for formatting the example docs, I have reverted the change on the compiler. I also agree it should be documented, and this becomes a forcing function to think about the problem when designing a solution that would replace the use-case

**Comment by medvednikov** [comment]
Ok, this can be documented, but only in the unsafe/C section, for working with very low level code or C code.

Pure V won't allow this.

**Comment by lilenvy** [comment]
Ok, I'll move it to the unsafe section, I'll squash the commits within the PR

I'm still interested in knowing if there's going to be a way of properly handling references as described here to do things like Linked Lists or Binary Trees

**Comment by lilenvy** [comment]
Alright, I nested it under "Memory-unsafe code", please let me know if I should change anything.

I would also like to keep the conversation going on things like reference fields and future plans, I started not liking V a lot on the beginning (rough corners!) but I'm liking it now and I'm happy to contribute to it. Is discord the only/best way for that or where should I go to have deeper conversation on this or other subjects?

Thanks

**Comment by medvednikov** [comment]
I squash them via github anyway.

Yes, discord is the best way.

File: doc/docs.md [hunk 1]
```diff
@@ -2737,6 +2737,41 @@ surrounding code).
...
 * Note: This is work in progress.
 
+### Structs with reference fields
+
+Structs with references require explicitly setting the initial value to a
+reference value unless the struct already defines its own initial value.
+
+Zero-value references, or nil pointers, will **NOT** be supported in the future,
+for now data structures such as Linked Lists or Binary Trees that rely on reference
+fields that can use the value `0` understanding it's unsafe and can casue a panic.
```
Comment (spytheman) [inline doc/docs.md:2747]: `casue` => `cause` or `lead to`

**Pull request merged**